packages/app-desktop/gui/TagList.tsx

```diff
@@ -29,9 +29,9 @@ function TagList(props: Props) {
 
 	const tags = useMemo(() => {
 		const output = props.items.slice();
-
+		const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' });
```

Comment:
does it not need to know the locale? looking at the MDN page, it has this example:
```
// in German, ä sorts with a
console.log(new Intl.Collator("de").compare("ä", "z"));
// -1, or some other negative value

// in Swedish, ä sorts after z
console.log(new Intl.Collator("sv").compare("ä", "z"));
// 1, or some other positive value
```

how does your change handle it?

Reply:
The original issue stated that the names were sorted correctly for Notes and the problem only occurred for Notebooks and Tags. 

Assuming that the sorting is fine for Notes, I took a similar approach here. Looking at the code for Notes, I don't think that the locale is being passed as a parameter to `Intl.Collator` during sorting, which is why I think your example wouldn't work for Notes either.

https://github.com/laurent22/joplin/blob/1b96a165866f3f9ad3499f45e01097e224797a27/packages/lib/models/Note.ts#L1121C2-L1123C3

        public static getNaturalSortingCollator() {
                return new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' });
        }


If we want to take the locale into consideration, the changes would have to be made for Notes as well, which I thought was out of the scope of this issue. 

If that is something that we actually want to do, I could try to make those changes test it with your example.

Reply:
Yeah, I'm not sure what's the right thing here.

From what I can see the original issue was only for Turkish language. It can be that the sorting method used for folders produces correct result for Turkish by accident but isn't correct for some other language. Unfortunately, I don't think there's an easy way to check, except maybe ask on the forum.

Reply:
Hey, I made some changes to the code to also consider the current locale before we do the sorting. I made it so that the notes, folders and tags are all sorted according to the current locale of the system. I tested for German, Swedish and some other languages and it seems to be working correctly, changing the sort order based on the locale. Some screenshots - 

For German, ä sorts with a - 

![german](https://github.com/laurent22/joplin/assets/91818868/6bbec76f-1e4b-4b16-b382-0123c79397e7)

For Swedish, ä sorts after z - 

![svenska](https://github.com/laurent22/joplin/assets/91818868/b0b7c09a-21df-4f8c-81c5-0b3b859fc307)

I think this approach makes more sense and is more inclusive than the previous commit.